google/cloud/bigquery/v2/minimal/mocks/mock_job_rest_stub.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_V2_MINIMAL_MOCKS_MOCK_JOB_REST_STUB_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_V2_MINIMAL_MOCKS_MOCK_JOB_REST_STUB_H
+
+#include "google/cloud/bigquery/v2/minimal/internal/job_rest_stub.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace bigquery_v2_minimal_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+/*
+ * A class to mock `BigQueryJobRestStub`.
+ *
+ * Application developers may want to test their code with simulated responses,
+ * including errors, from an object of type `BigQueryJobConnection`. To do so,
+ * construct an object of type `BigQueryJobConnection` with an instance of this
+ * class. Then use the Google Test framework functions to program the behavior
+ * of this mock.
+ */
+// TODO(#11039): Move to "bigquery_v2_minimal_mocks" namespace when api is ready
+// to go public.
```

Comment:
FYI: These mock stubs are something we use to test our internal code. But they are not something our customers will need. We tend to put them in `*_testing` libraries. (Whereas the `*_mocks` libraries are intended only for customer facing mocks).

No action needed from you. I will send a PR to shuffle these things around. Your work may be slightly affected, as I will change this filepath, and you will likely make additions to the file. (It will be a race to see who gets their PR out first)

Reply:
Thanks Darren! I am currently working on a different api so hoefully there won't be any conflict.

Please let me know whenever these have been merged and I'll pull them. Thanks!